server/app/views/admin/programs/ProgramBlockPredicateConfigureView.java

```diff
@@ -0,0 +1,621 @@
+                            ReferenceClasses.MULTI_OPTION_VALUE)
+                        .withCondSelected(
+                            maybeExistingPredicate
+                                .map(PredicateDefinition::action)
+                                .map(existingAction -> existingAction.equals(action))
+                                .orElse(false))
+                        .withValue(action.name())));
+
+    container.with(selectTag);
+
+    return container;
+  }
+
+  private ImmutableList<PredicateExpressionNode> getExistingAndNodes(
+      PredicateDefinition existingPredicate) {
+    switch (existingPredicate.computePredicateFormat()) {
+      case SINGLE_QUESTION:
+        {
+          return ImmutableList.of(
+              PredicateExpressionNode.create(
+                  AndNode.create(ImmutableList.of(existingPredicate.rootNode()))));
+        }
+
+      case OR_OF_SINGLE_LAYER_ANDS:
+        {
+          return existingPredicate.rootNode().getOrNode().children();
+        }
+
+      default:
+        {
+          throw new RuntimeException(
+              String.format(
+                  "Unrecognized predicate format: %s", existingPredicate.computePredicateFormat()));
+        }
+    }
+  }
+
+  private DivTag renderQuestionHeaders(
+      ImmutableList<QuestionDefinition> questionDefinitions,
+      Optional<PredicateDefinition> maybeExistingPredicate) {
+    DivTag container = div().withClasses("flex", "py-4");
+    int columnNumber = 1;
+
+    if (maybeExistingPredicate.isPresent()) {
+      ImmutableList<LeafOperationExpressionNode> leafNodes =
+          getExistingAndNodes(maybeExistingPredicate.get()).stream()
+              .findFirst()
+              .get()
+              .getAndNode()
+              .children()
+              .stream()
+              .map(PredicateExpressionNode::getLeafNode)
+              .sorted(
+                  (LeafOperationExpressionNode loenA, LeafOperationExpressionNode loenB) ->
+                      (int) (loenA.questionId() - loenB.questionId()))
+              .collect(ImmutableList.toImmutableList());
+
+      for (LeafOperationExpressionNode leafNode : leafNodes) {
+        long questionId = leafNode.questionId();
+        QuestionDefinition qd =
+            questionDefinitions.stream()
+                .filter(questionDefinition -> questionDefinition.getId() == questionId)
+                .findFirst()
+                .get();
+
+        container.with(
+            div(
+                    div(qd.getQuestionText().getDefault())
+                        .withClasses(BaseStyles.INPUT, "text-gray-500", "mb-2", "truncate"),
+                    createScalarDropdown(qd, Optional.of(leafNode.scalar())),
+                    createOperatorDropdown(
+                        qd, Optional.of(leafNode.scalar()), Optional.of(leafNode.operator())))
+                .withClasses(COLUMN_WIDTH, columnNumber++ != 1 ? "ml-16" : null));
```

Comment:
```suggestion
                .withClasses(COLUMN_WIDTH, iff(columnNumber++ != 1, "ml-16")));
```

Reply:
This should work.... though I don't see that we have other examples of it.

Reply:
done